### PR TITLE
tailcfg,hostinfo: add Hostinfo.Machine and Hostinfo.GoArchVar

### PR DESCRIPTION
**Pull request by bradfitz**
For detecting a non-ideal binary running on the current CPU.

And for helping detect the best Synology package to update to.

Updates #6995

### CODE REVIEW

File: tailcfg/tailcfg.go [hunk 1]
```diff
@@ -528,7 +528,9 @@ type Hostinfo struct {
 	ShareeNode      bool           `json:",omitempty"` // indicates this node exists in netmap because it's owned by a shared-to user
 	NoLogsNoSupport bool           `json:",omitempty"` // indicates that the user has opted out of sending logs and support
 	WireIngress     bool           `json:",omitempty"` // indicates that the node wants the option to receive ingress connections
+	Machine         string         `json:",omitempty"` // the host's machine type (uname -m)
 	GoArch          string         `json:",omitempty"` // the host's GOARCH value (of the running binary)
+	GoArchVar       string         `json:",omitempty"` // GOARM, GOAMD64, etc
```
Comment (DentonGentry) [inline tailcfg/tailcfg.go:533]: The previous comment says it is the running binary, I think this one is the compiled binary?

File: hostinfo/hostinfo.go [hunk 2]
```diff
@@ -321,6 +326,40 @@ func inDockerDesktop() bool {
 	return false
 }
 
+// goArchVar returns the GOARM or GOAMD64 etc value that the binary was built
+// with.
+func goArchVar() string {
```
Comment (DentonGentry) [inline hostinfo/hostinfo.go:331]: If a GOARM64 or GORISCV is ever added I don't think we have a way to automatically detect it, I don't see a way to check https://go.dev/src/internal/buildcfg/cfg.go in a CI test or something.

**Review by DentonGentry — APPROVED** [review]

File: tailcfg/tailcfg.go (hunk 1 above)
Comment (bradfitz) [reply tailcfg/tailcfg.go:533]: clarified all

**Review by DentonGentry — APPROVED** [review]

File: hostinfo/hostinfo.go (hunk 2 above)
Comment (bradfitz) [reply hostinfo/hostinfo.go:331]: Okay, I rewrote it to be future-proof, as long as the future follows the conventions thus far. Not that we really care about anything except GOARM, which is how this all started 😅

**Pull request merged**